test/test_creating_dataset.py

```diff
@@ -30,6 +39,175 @@ def test_is_path_correct():
     assert is_path_correct(DATASET_PATHS[0])
 
 
+def test_split_characteristics_into_groups():
+    test_speeds = {"fly"}
+    test_weaknesses = {"good_weakness"}
+    test_resistances = {"fire_resistance"}
+    test_special_characteristics = {"ranged"}
+    test_characteristics_rename = {"num_immunities", "int"}
+
+    input_characteristics = {
+        "wrong_one",
+        "fly",
+        "good_weakness",
+        "fire_resistance",
+        "ranged",
+        "num_immunities",
+        "int",
+    }
+    characteristics_groups = split_characteristics_into_groups(input_characteristics)
+    (
+        speeds,
+        weaknesses,
+        resistances,
+        special_characteristics,
+        characteristics_rename,
+    ) = (
+        characteristics_groups.speeds,
+        characteristics_groups.weaknesses,
+        characteristics_groups.resistances,
+        characteristics_groups.special_characteristics,
+        characteristics_groups.characteristics_rename,
+    )
+
+    assert (
+        speeds,
+        weaknesses,
+        resistances,
+        special_characteristics,
+        characteristics_rename,
+    ) == (
+        test_speeds,
+        test_weaknesses,
+        test_resistances,
+        test_special_characteristics,
+        test_characteristics_rename,
+    )
```

Comment:
Proponuję to rozpisać na osobne asserty, bo jeżeli byłby tu błąd np. z resistances, to stacktrace byłby bardzo nieczytelny